src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java

```diff
@@ -2,35 +2,38 @@
 
 import com.google.common.collect.ImmutableMap;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
-import org.springframework.stereotype.Component;
 import org.springframework.stereotype.Repository;
 import org.springframework.transaction.annotation.Transactional;
 import uk.ac.ebi.atlas.experimentpage.tsne.TSnePoint;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Repository
 public class TSnePlotDao {
     private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    private final String TSNE_METHOD = "tsne";
 
     public TSnePlotDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate) {
         this.namedParameterJdbcTemplate = namedParameterJdbcTemplate;
     }
 
     private static final String SELECT_T_SNE_PLOT_WITH_EXPRESSION_STATEMENT =
-            "SELECT tsne.cell_id, tsne.x, tsne.y, analytics.expression_level " +
-                    "FROM scxa_tsne AS tsne " +
-                    "LEFT JOIN " +
-                    "(SELECT * FROM scxa_analytics WHERE gene_id=:gene_id and experiment_accession=:experiment_accession) AS analytics " +
-                    "ON analytics.cell_id=tsne.cell_id " +
-                    "WHERE tsne.experiment_accession=:experiment_accession AND tsne.perplexity=:perplexity";
+        "SELECT c.cell_id, c.x, c.y, a.expression_level " +
+                "FROM scxa_coords c LEFT JOIN scxa_analytics a ON " +
+                "c.experiment_accession=a.experiment_accession AND " +
+                "c.cell_id=a.cell_id AND a.gene_id=:gene_id " +
+                "WHERE c.experiment_accession=:experiment_accession AND " +
+                "c.method=:method AND c.parameterisation->0->>'perplexity'=:perplexity " +
+                "ORDER BY c.cell_id";
```

Comment:
Sorry, this is my mistake- you probably don't need the sorting by cell ID (at least the query you gave me didn't have it I think)- I just used it for testing and comparison. The sort may slow things down so you should remove it unless you need it.